### ISSUE TITLE
Add code auto-formatting to repo

### ISSUE THREAD

**Issue by sea-bass** [PlanSys2/ros2_planning_system#273]
This is something I'm happy to work on by adding in [pre-commit](https://pre-commit.com/) plus a CI action to check auto-formatting.

However, before I try this I wanted to get the maintainers' go-ahead.

**Comment by fmrico** [comment]
Hi @sea-bass 

CI is already passing the usual tests, like cpplint and uncrustify. If it does the same? What more does your proposal contribute to CI? Another concern is to have the opportunity for contributors to pass the same repo CI tests in their machines.

**Comment by fmrico** [comment]
Is this related to #248?

**Comment by sea-bass** [comment]
Yes it is, and didn't realize there was already a closed PR. My apologies.

I like pre-commit because it extends beyond just C++ code -- we use it to format Python, YAML, Dockerfiles, markdown docs, etc. And it also supports running `clang-format` on C++.

cpplint and uncrustify are a little more work to run, and most contributors (like myself) don't realize it's even there in CI until I fail some checks.

The `CONTRIBUTING.md` should definitely mention how to run these locally as part of commit requirements. Just a suggestion.

**Comment by robodrome** [comment]
> Yes it is, and didn't realize there was already a closed PR. My apologies.
> 
> I like pre-commit because it extends beyond just C++ code -- we use it to format Python, YAML, Dockerfiles, markdown docs, etc. And it also supports running `clang-format` on C++.
> 
> cpplint and uncrustify are a little more work to run, and most contributors (like myself) don't realize it's even there in CI until I fail some checks.
> 
> The `CONTRIBUTING.md` should definitely mention how to run these locally as part of commit requirements. Just a suggestion.

I agree. I had some issues with formatting.
